### PR TITLE
Add support for adding config through class kwargs

### PR DESCRIPTION
**Pull request by hramezani**
Closes https://github.com/pydantic/pydantic-settings/issues/110

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by hramezani** [comment]
please review

File: pydantic_settings/main.py [hunk 1]
```diff
@@ -27,6 +28,11 @@ class SettingsConfigDict(ConfigDict, total=False):
     secrets_dir: str | Path | None
 
 
+# Extend config_keys by pydantic settings config keys to
+# support setting config through class kwargs.
+config_keys |= set(SettingsConfigDict.__annotations__.keys())
```
Comment (Kludex) [inline pydantic_settings/main.py:33]: This seems super magical... Can you add a reference to the pydantic code that we are using here?

**Review by Kludex — CHANGES_REQUESTED** [review]

File: pydantic_settings/main.py (hunk 1 above)
Comment (hramezani) [reply pydantic_settings/main.py]: I improved the comment and explained it more by mentioning the class and method name that the magic happen there on Pydantic.

Please let me know by comment or suggestion if you think it is not clear yet.

**Review by Kludex — APPROVED** [review]

**Comment by jordantshaw** [comment]
Which release is this going with? @hramezani

**Comment by hramezani** [comment]
> Which release is this going with? @hramezani

Next release will happen soon and this will be included

**Pull request merged**